Nine-point circle

enedil opened the pull request:
https://en.wikipedia.org/wiki/Nine-point_circle
Create nine-point circle

enedil committed:
Nine-point circle

https://en.wikipedia.org/wiki/Nine-point_circle
Create nine-point circle

smichr commented:
+1

leosartaj commented on sympy/geometry/polygon.py:
```diff
@@ -2249,6 +2250,40 @@ def medial(self):
+        return Circle(*self.medial.vertices)
+
+
+
```
You can remove two extra empty lines here.

leosartaj commented:
You should add few tests for the new addition.

leosartaj commented:
Other than my few comments. This looks good. +1

enedil committed:
Tests+fixed formatting

First test checks whether the circle passes through the middle points
of segments connecting the vertices with orthocenter, second one is
just a static check for being the circumcircle of medial triangle

enedil committed:
delete space

enedil commented:
Ok, apparently fixed those.
Are you interested in implementation of other triangle centers?

leosartaj commented on sympy/geometry/polygon.py:
```diff
@@ -2249,6 +2250,38 @@ def medial(self):
+        >>> p1, p2, p3 = Point(0, 0), Point(1, 0), Point(0, 1)
+        >>> t = Triangle(p1, p2, p3)
+        >>> t.nine_point_circle
+        Circle(Point(1/4, 1/4), sqrt(2)/4)
```
``Circle(Point2D(1/4, 1/4), sqrt(2)/4)``

leosartaj commented on sympy/geometry/tests/test_polygon.py:
```diff
@@ -228,6 +228,11 @@ def test_polygon():
     assert t1.medial == Triangle(Point(2.5, 0), Point(0, 2.5), Point(2.5, 2.5))
 
+    # Nine-point circle
+    vo_middles = [Segment(v, t1.orthocenter).midpoint for v in t1.vertices]
```
``vo_middles`` in this case is ``[Point2D(2, 0), <bound method Point2D.midpoint of Point2D(4, 0)>, Point2D(4, 3/2)]``, probably due to t1 being a right-angle triangle.

enedil replied:
@leosartaj Yeah, since `t1` is both right and isosceles, 9 points are degenerated to only 4 distinct.

leosartaj replied:
You have to probably change this line, as this causes test failure.

enedil replied:
Yeah, I even did commit, albeit did not pull.

enedil committed:
Changes due to specific triangle used in test

leosartaj commented:
This looks good. Merging.

leosartaj commented:
Thanks @enedil !
